### DISCUSSION TITLE
eth recharge

### DISCUSSION THREAD

**Discussion by Poetry-Distance** [zkSync-Community-Hub/zksync-developers#687] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

How many blocks are needed for confirmation or how long it takes to be on chain to confirm that an ETH recharge will not be rolled back

**Comment by alhenaai** [comment]
For an Ethereum (ETH) deposit to ZKsync, the transaction typically requires about 10 block confirmations on the Ethereum L1 network. This process usually takes around 5 minutes, but the exact time can vary depending on the congestion of the Ethereum network and the L1 gas fee set for the transaction.

Once these confirmations are completed, the transaction is considered secure and highly unlikely to be rolled back. If you have any more specific details or concerns about a transaction, feel free to provide the transaction hash or further context!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply]
This post might give you a little insight to what you're asking https://github.com/zkSync-Community-Hub/zksync-developers/discussions/616

When you say recharge, what do you mean? Like a deposit from L1 to ZKsync Era?

**Reply by Poetry-Distance** [reply]
Does the RPC field returned by zksync indicate how many times this field has been confirmed on L1?
If not, what is the effective logic for the maturity confirmation of zksync transactions that you recommend? As a centralized exchange, we want to confirm the legality of users' recharge transactions based on this logic.

**Reply by uF4No** [reply, marked as answer]
> Does the RPC field returned by zksync indicate how many times this field has been confirmed on L1?

No but [`zks_getTransactionDetails`](https://docs.zksync.io/build/api-reference/zks-rpc#zks_gettransactiondetails) will give you the status of the tx ([more info in our docs](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle#transaction-statuses)). If it's verified, it's settled on Ethereum.
